### PR TITLE
Fix flaky testCreateApiKey test

### LINKED ISSUE #47958: [CI] ApiKeyIntegTests.testCreateApiKey fails rarely
The test ApiKeyIntegTests.testCreateApiKey failed in 
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob+fast+part2/1774/console

I could not reproduce this locally with:
```
./gradlew ':x-pack:plugin:security:test' \
--tests "org.elasticsearch.xpack.security.authc.ApiKeyIntegTests.testCreateApiKey" \
-Dtests.seed=189C6A8AC15D3889 \
-Dtests.security.manager=true \
-Dtests.locale=sd \
-Dtests.timezone=America/Rosario 
-Dcompiler.java=12 \
-Druntime.java=11
```

### PR DESCRIPTION
**Pull request by jkakavas**
API Key expiration value has millisecond precision as we use
{@link Instant#toEpoqueMilli()} when creating the API key
document.
It could often happen that `Instant.now()` Instant in the testCreateApiKey
was close enough to the ApiKeyService's `clock.instant()` Instant,
when the nanos were removed from the latter ( due to the call
to `toEpoqueMilli()` ) the result of comparing these two Instants
was a few nanos short of a 7 days.

Resolves: #47958

### CODE REVIEW

**Comment by jkakavas** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by jkakavas** [comment]
@elastimachine update branch

**Comment by jkakavas** [comment]
@elastimachine update branch

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java [hunk 1]
```diff
@@ -131,7 +131,6 @@ private void awaitApiKeysRemoverCompletion() throws Exception {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/47958")
     public void testCreateApiKey() throws Exception{
         final Instant start = Instant.now();
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java]:
```suggestion
         final Instant start = Instant.ofEpochMilli(Instant.now().toEpochMilli());
```

**Review by albertzaharovits — APPROVED** [review]
Nice catch! I raised a stylistic nit comment, LGTM to merge with/without it.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java (hunk 1 above)
Comment (jkakavas) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java]: I like that, I'll keep it and remove the nano removing part below

**Pull request merged**